wagtail/wagtailadmin/templates/wagtailadmin/shared/breadcrumb.html

```diff
@@ -1,8 +1,18 @@
 {% load i18n wagtailadmin_tags %}
 
 <ul class="breadcrumb">
-    {% for page in page.get_ancestors %}
-        {% if page.is_root %}
+    {% if choosing %}
+        {% current_users_choosable_ancestors page request as ancestors %}
+    {% else %}
+        {% current_users_explorable_ancestors page request as ancestors %}
+    {% endif %}
+    {% for page in ancestors %}
+        {% if choosing %}
+            {% is_current_users_choosable_root page request as page_is_permitted_root %}
+        {% else %}
+            {% is_current_users_explorable_root page request as page_is_permitted_root %}
+        {% endif %}
+        {% if page_is_permitted_root %}
```

Comment:
Having this extent of non-visual logic in the template probably indicates that breadcrumb.html itself should become a template tag rather than just an include, so that we can do this in Python code. I'll have a go at refactoring this.